### PR TITLE
_plots.py: use commit_repo_url prop, fix #1483

### LINKED ISSUE #1483: res-res cmp:  `TypeError: 'NoneType' object is not subscriptable`
```
unexpected exception, please report this:
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/views.py", line 107, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/views.py", line 188, in dispatch_request
    return current_app.ensure_sync(meth)(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/conbench/app/_endpoint.py", line 28, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/app/conbench/app/compare.py", line 204, in get
    return self.page(
           ^^^^^^^^^^
  File "/app/conbench/app/compare.py", line 72, in page
    benchmark_result_history_plot_info = self.get_history_plot(
                                         ^^^^^^^^^^^^^^^^^^^^^^
  File "/app/conbench/app/_plots.py", line 128, in get_history_plot
    time_series_plot(
  File "/app/conbench/app/_plots.py", line 680, in time_series_plot
    repo_string=run["commit"]["repository"]
                ~~~~~~~~~~~~~^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable
```

Seen in prod.

### PR DESCRIPTION
**Pull request by jgehrcke**
For addressing #1483.

Things get better and better. `repo_string=run["commit"]["repository"]` contained ambiguity and was bound to fail at some point; this line of code contains more than one weakness that we had been working on in the past; and it's good that as of today we can replace it with a more well-defined alternative: `repo_url=current_benchmark_result.commit_repo_url`.

### CODE REVIEW

File: conbench/app/_plots.py [hunk 1]
```diff
@@ -450,13 +452,19 @@ def gen_js_callback_tap_detect_unselect(source: bokeh.models.ColumnDataSource):
...
     """
     Dynamically manipulate Bootstrap panel conbench-histplot-run-details`.
 
     Show run details if a corresponding glyph was clicked in the plot.
 
     Warning: requires manual testing, not covered by tests/CI.
+
+    Note(JP): `repo_url` really is expected to be a URL and we have done quite
+    a bit of work in the last months to drive towards consistency, but at the
+    time of writing this sentence I am not sure if it really always is a URL.
```
Comment (austin3dickey) [inline conbench/app/_plots.py]: Yes, this is guaranteed to always be a URL.

https://github.com/conbench/conbench/blob/cf7931fbb415809bc3651f7f74692e09068468ac/conbench/entities/benchmark_result.py#L86

**Review by austin3dickey — APPROVED** [review]
Thank you!

File: conbench/app/_plots.py (hunk 1 above)
Comment (jgehrcke) [reply conbench/app/_plots.py]: Ha, I could have known! :) Thanks, adjusted comment.

File: conbench/app/_plots.py [hunk 2]
```diff
@@ -905,6 +910,13 @@ def get_source_for_single_benchmark_result(current_benchmark_result, cur_run, un
     # source_current_bm_min = None
     # source_current_bm_mean = None
 
+    # TODO: `cur_run["commit"]` can be None, I think, and that means that
+    # cur_run["commit"]["timestamp"] may fail. Type checking does not know this
+    # because those dictionaries are untyped. `cur_run` I think is here only to
+    # communicate commit information. Once commit information is stores more
+    # clearly on the benchmark result object we can remove the `cur_run`
+    # argument and have type checking find potential bugs.
```
Comment (austin3dickey) [inline conbench/app/_plots.py:918]: ah, yeah you were right. https://github.com/conbench/conbench/issues/1487

**Pull request merged**